### PR TITLE
Ports Shoc's isotropic_ts to C++/Kokkos

### PR DESCRIPTION
**Pull request by singhbalwinder**
Added BFB test as well

### CODE REVIEW

File: components/scream/src/physics/shoc/shoc_integ_column_stability_impl.hpp [hunk 1]
```diff
@@ -18,7 +18,7 @@ ::integ_column_stability(
   Scalar&                      brunt_int)
 {
   //Lower troposphere pressure [Pa]
-  static constexpr auto troppres = 80000;
+  static constexpr Scalar troppres = 80000;
```
Comment (singhbalwinder) [inline components/scream/src/physics/shoc/shoc_integ_column_stability_impl.hpp:21]: @ambrad suggested this change in my previous attempt at porting this function.

File: components/scream/src/physics/shoc/shoc_isotropic_ts_impl.hpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  const uview_1d<const Spack>& tke,
+  const uview_1d<const Spack>& a_diss,
+  const uview_1d<const Spack>& brunt,
+  const uview_1d<Spack>&       isotropy)
+{
+
+  //constants from physics/share
+  static constexpr  auto ggr = C::gravit;
+
+  //Declate constants as Spacks to use ekat's min and max functions
```
Comment (singhbalwinder) [inline components/scream/src/physics/shoc/shoc_isotropic_ts_impl.hpp:31]: This should be looked at carefully. I declared them as Spacks as I wanted to use ekat's min and max functions. I can modify the code if there is a better way to do it.

**Review by jeff-cohere — APPROVED** [review]

File: components/scream/src/physics/shoc/shoc_isotropic_ts_impl.hpp [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  const uview_1d<const Spack>& a_diss,
+  const uview_1d<const Spack>& brunt,
+  const uview_1d<Spack>&       isotropy)
+{
+
+  //constants from physics/share
+  static constexpr  Scalar ggr = C::gravit;
+
+  //Declare constants
+  static constexpr Scalar lambda_low   = sp(0.001);
```
Comment (jgfouca) [inline components/scream/src/physics/shoc/shoc_isotropic_ts_impl.hpp:32]: These's sp's aren't necessary.

File: components/scream/src/physics/shoc/tests/shoc_tke_isotropic_ts_tests.cpp [hunk 4]
```diff
@@ -175,8 +175,55 @@ struct UnitWrap::UnitTest<D>::TestShocIsotropicTs {
 
   static void run_bfb()
   {
-    // TODO
-  }
+    SHOCIsotropicData f90_data[] = {
```
Comment (jgfouca) [inline components/scream/src/physics/shoc/tests/shoc_tke_isotropic_ts_tests.cpp:178]: This struct has been regenerated with a new name, so so minor changes will be needed here.

File: components/scream/src/physics/shoc/tests/shoc_tke_isotropic_ts_tests.cpp [hunk 5]
```diff
@@ -175,8 +175,55 @@ struct UnitWrap::UnitTest<D>::TestShocIsotropicTs {
...
+    // Get data from fortran
+    for (auto& d : f90_data) {
+      // expects data in C layout
+      isotropic_ts(d);
+    }
+
+    // Get data from cxx
+    for (auto& d : cxx_data) {
+      d.transpose<ekat::TransposeDirection::c2f>(); // _f expects data in fortran layout
+      isotropic_ts_f(d.nlev(), d.shcol(), d.brunt_int, d.tke, d.a_diss, d.brunt, d.isotropy);
```
Comment (jgfouca) [inline components/scream/src/physics/shoc/tests/shoc_tke_isotropic_ts_tests.cpp:213]: nlev, shcol are no longer methods.

File: components/scream/src/physics/shoc/tests/shoc_tke_isotropic_ts_tests.cpp [hunk 6]
```diff
@@ -175,8 +175,55 @@ struct UnitWrap::UnitTest<D>::TestShocIsotropicTs {
...
+      d.transpose<ekat::TransposeDirection::f2c>(); // go back to C layout
+    }
+
+    // Verify BFB results, all data should be in C layout
+    for (Int i = 0; i < num_runs; ++i) {
+      SHOCIsotropicData& d_f90 = f90_data[i];
+      SHOCIsotropicData& d_cxx = cxx_data[i];
+      for (Int k = 0; k < d_f90.total1x2(); ++k) {
```
Comment (jgfouca) [inline components/scream/src/physics/shoc/tests/shoc_tke_isotropic_ts_tests.cpp:221]: use `d_f90.total(isotropy)`

**Review by jgfouca — CHANGES_REQUESTED** [review]
Please do an upstream merge. I've marked the places that will need to change when you do it.

**Review by xyuan — APPROVED** [review]
It sounds good to me if Jame's suggestion is addressed.

**Comment by singhbalwinder** [comment]
Thank you all! I have all the review comments.

**Review by jgfouca — APPROVED** [review]

**Review by tcclevenger — APPROVED** [review]

**Review by tcclevenger — APPROVED** [review]

**Pull request merged**